android/src/main/java/com/appcuesreactnative/AppcuesFrameViewManager.kt

```diff
@@ -0,0 +1,78 @@
+package com.appcuesreactnative
+
+import android.content.Context
+import android.view.View
+import android.widget.FrameLayout
+import androidx.core.view.children
+import com.appcues.AppcuesFrameView
+import com.facebook.react.uimanager.ThemedReactContext
+import com.facebook.react.uimanager.UIManagerModule
+import com.facebook.react.uimanager.ViewGroupManager
+import com.facebook.react.uimanager.annotations.ReactProp
+
+
+internal class AppcuesFrameViewManager: ViewGroupManager<AppcuesWrapperView>() {
+
+    override fun getName() = REACT_CLASS
+
+    companion object {
+        const val REACT_CLASS = "AppcuesFrameView"
+    }
+
+    override fun createViewInstance(context: ThemedReactContext): AppcuesWrapperView {
+        val wrapperView = AppcuesWrapperView(context)
+        val frameView = AppcuesFrameView(context)
+        wrapperView.contentView = frameView
+        return wrapperView
+    }
+
+    @ReactProp(name = "frameID")
+    fun setFrameId(view: AppcuesWrapperView, frameId: String) {
+        (view.contentView as? AppcuesFrameView)?.let {
+            AppcuesReactNativeModule.implementation?.registerEmbed(frameId, it)
+        }
+    }
+}
+
+internal class AppcuesWrapperView(context: Context) : FrameLayout(context) {
+    var contentView: View? = null
```

Comment:
Can the type of this be `AppcuesFrameView`? Would let you remove the `as?` above.

Reply:
yep, will clean that up 👍